Cython/Utils.py

```diff
@@ -264,15 +265,23 @@ def check_package_dir(dir_path, package_names):
 @cached_function
 def contains_init(dir_path):
     for filename in PACKAGE_FILES:
-        path = os.path.join(dir_path, filename)
-        if path_exists(path):
+        paths = glob.glob(os.path.join(dir_path, filename))
+        if paths:
             return 1
 
-
 def is_package_dir(dir_path):
     if contains_init(dir_path):
         return 1
 
+def is_package_file(file_path):
+    if file_path is None:
+        return False
+    filename = os.path.basename(file_path)
+    for package_file in PACKAGE_FILES:
+        regex = fnmatch.translate(package_file)
+        if re.match(regex, filename):
```

Comment:
I am not sure whether compiling everytime regex can have performance impact or not. If yes what is better approach cache function or precompile the PACKAGE_FILES? Or both? :-)

Reply:
For now, I cached the `is_package_file()` function.

Reply:
I think practically the `re` module does some internal caching of compiled regexes itself. It's probably less good than precompiling `PACKAGE_FILES` but it probably isn't as bad as it seems at first

Reply:
Yes, regex compilation should be considered cheap.